docs/root/intro/arch_overview/upstream/aggregate_cluster.rst

```diff
@@ -0,0 +1,61 @@
+.. _arch_overview_aggregate_cluster:
+
+Aggregate Cluster
+=================
+
+Aggregate cluster is used for failover between clusters from different service discovery mechanism, e.g., 
+failover from EDS upstream cluster to STRICT_DNS upstream cluster. Aggregate cluster loosely couples multiple 
+clusters by referencing their name in the :ref:`configuration <envoy_api_msg_config.cluster.aggregate.ClusterConfig>`. 
+The fallback priority is defined implicitly by the ordering in the :ref:`clusters <envoy_api_field_config.cluster.aggregate.ClusterConfig.clusters>`.
```

Comment:
Or explicit now?

Reply:
Not implemented yet. We can update the document once it's implemented.

Reply:
What will happen to the ordering property when we have explicit priorities? It no longer figures? Ideally, we would have the existing behavior as the degenerate fallback when all the weights are the same.

Reply:
That's what I thought. When priority exists, clusters are ordered by the priority; when no priority (or priorities are same), clusters are ordered by the ordering in the list.

Reply:
Wouldn't a degenerate behavior be that they all have the same order/weight?

Reply:
1. If no priority specified, use ordering in the list
2. If priorities are all specified and are all different, use priorities
3. If priorities are all specified and are all same, use the ordering in the list
4. If priorities are all specified and some are same, ordered by priority + ordering in the list for clusters with same priority. <- Implementation details
5. If priorities are partially specified, ... Maybe we don't need to support this case.

Reply:
If we want different behavior between the two should we just use a `oneof` so that its clearer that these are two different ways of configuring the cluster?

Reply:
Can we make the `priority` as required filed once it's supported?